app/electron/main.ts

```diff
@@ -1087,6 +1156,72 @@ async function getRunningHeadlampPIDs() {
   return processes.map(pInfo => pInfo.pid);
 }
 
+/**
+ * Check if a specific port is occupied by a Headlamp process
+ * @returns Array of Headlamp PIDs using the port, or null if port is free or used by another process
+ */
+async function getHeadlampPIDsOnPort(port: number): Promise<number[] | null> {
+  try {
+    // First check if the port is in use at all
+    const portInUse = !(await isPortAvailable(port));
+    if (!portInUse) {
+      console.debug(`Port ${port} is available (not in use)`);
+      return null; // Port is free
+    }
+
+    console.debug(`Port ${port} is in use, checking if it's Headlamp...`);
+
+    // Get all Headlamp processes
+    const headlampProcesses = await find_process('name', 'headlamp-server');
+    console.debug(
+      `Found ${headlampProcesses.length} Headlamp process(es):`,
+      headlampProcesses.map(p => ({ pid: p.pid, name: p.name, cmd: p.cmd }))
+    );
+
+    if (headlampProcesses.length === 0) {
+      console.debug(`No Headlamp processes running`);
+      return null; // No Headlamp processes running
+    }
+
+    // Parse command line arguments to find which Headlamp process is using this port
+    const headlampOnPort = headlampProcesses.filter(p => {
```

Comment:
Using the process list for this doesn't work for one case. Where you open Headlamp on Windows, then open Headlamp on WSL. (Or any other VM system on other OSes). This is because the process list is not available between the two OSes.